pkg/kubelet/dockershim/security_context.go

```diff
@@ -68,16 +76,29 @@ func applyContainerSecurityContext(lc *runtimeapi.LinuxContainerConfig, podSandb
 }
 
 // modifyContainerConfig applies container security context config to dockercontainer.Config.
-func modifyContainerConfig(sc *runtimeapi.LinuxContainerSecurityContext, config *dockercontainer.Config) {
+func modifyContainerConfig(sc *runtimeapi.LinuxContainerSecurityContext, config *dockercontainer.Config) error {
 	if sc == nil {
-		return
+		return fmt.Errorf("LinuxContainerSecurityContext is nil")
 	}
 	if sc.RunAsUser != nil {
 		config.User = strconv.FormatInt(sc.GetRunAsUser().Value, 10)
 	}
 	if sc.RunAsUsername != "" {
 		config.User = sc.RunAsUsername
 	}
+
+	user := config.User
+	if sc.RunAsGroup != nil {
+		if user == "" {
+			return fmt.Errorf("Group is specified without a User, %s", sc.RunAsGroup)
+		}
+		gid := strconv.FormatInt(sc.GetRunAsGroup().Value, 10)
+		user = fmt.Sprintf("%s:%s", config.User, gid)
```

Comment:
It's kind of unfortunate that you can't specify a group unless you also specify a user.

Reply:
yep i tested thats the behavior in docker as well

Reply:
do we check that in validation (must set runAsUser if you set runAsGroup)?